src/test/resources/test-rom/spec/modules/cc/strings_spec.lua

```diff
@@ -0,0 +1,42 @@
+local with_window = require "test_helpers".with_window
+
+describe("cc.pretty", function()
+    local str = require("cc.strings")
+
+    describe("wrap", function()
+        it("validates arguments", function()
+            str.wrap("test string is long")
+            str.wrap("test string is long", 11)
+            expect.error(str.wrap, nil):eq("bad argument #1 (expected string, got nil)")
+            expect.error(str.wrap, "", false):eq("bad argument #2 (expected number, got boolean)")
+        end)
+
+        it("wraps lines", function()
+            expect(str.wrap("test string is long")[1]):eq("test string is long")
+
+            expect(str.wrap("test string is long", 15)[1]):eq("test string is ")
+            expect(str.wrap("test string is long", 15)[2]):eq("long")
+
+            expect(str.wrap("test string is long", 12)[1]):eq("test string ")
+            expect(str.wrap("test string is long", 12)[2]):eq("is long")
+
+            expect(str.wrap("test string is long", 11)[1]):eq("test string")
```

Comment:
This test fails, I can't figure out how to remove the trailing space nicely.

Reply:
We only take the length of the line into account when appending new words, not spaces. So `test string` is fine, then we see a space (`test string `) and _then_ we add a new line.

The normalisation code above (`:sub(1, width)`) should fix that problem, hence me suggesting keeping that but not the padding with additional spaces :p.